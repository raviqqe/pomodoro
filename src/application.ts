export type * from "./application/performance-graph.js";
