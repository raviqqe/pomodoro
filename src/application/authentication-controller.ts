export interface IAuthenticationController {
  signIn(): Promise<boolean>;
  signOut(): Promise<boolean>;
  isSignedIn(): Promise<boolean>;
}
