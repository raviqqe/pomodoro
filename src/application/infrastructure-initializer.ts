export interface IInfrastructureInitializer {
  initialize(): Promise<void>;
}
