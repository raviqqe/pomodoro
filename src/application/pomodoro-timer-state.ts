export enum PomodoroTimerState {
  Pomodoro,
  ShortBreak,
  LongBreak,
}
