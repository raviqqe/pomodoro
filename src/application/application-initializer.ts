import { IAuthenticationController } from "./authentication-controller";
import { IInfrastructureInitializer } from "./infrastructure-initializer";

export interface IInitialState {
  signedIn: boolean;
}

export class ApplicationInitializer {
  constructor(
    private readonly infrastructureInitializer: IInfrastructureInitializer,
    private readonly authenticationController: IAuthenticationController
  ) {}

  public async initialize(): Promise<IInitialState> {
    await this.infrastructureInitializer.initialize();
    return { signedIn: await this.authenticationController.isSignedIn() };
  }
}
