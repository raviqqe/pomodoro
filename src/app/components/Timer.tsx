import { styled } from "@linaria/react";
import { useStore } from "@nanostores/react";
import { PomodoroTimerState } from "../../application/pomodoro-timer-state.js";
import { pomodoroTimerPresenter } from "../../main/pomodoro-timer-presenter.js";
import { pomodoroTimerStarter } from "../../main/pomodoro-timer-starter.js";
import { pomodoroTimerStopper } from "../../main/pomodoro-timer-stopper.js";
import { white } from "../style.js";
import { TextButton } from "./TextButton.js";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  > * {
    margin: 1rem;
  }
`;

const State = styled.div`
  font-size: 5em;
`;

const Time = styled.div`
  color: ${white};
  font-size: 10em;
`;

const Minutes = styled.span``;

const Seconds = styled.span`
  font-size: 0.6em;
  margin-left: 0.2ex;
`;

export const Timer = (): JSX.Element => {
  const stopped = useStore(pomodoroTimerPresenter.stopped);
  const state = useStore(pomodoroTimerPresenter.state);
  const seconds = useStore(pomodoroTimerPresenter.seconds);

  return (
    <Container>
      {stopped ? (
        <>
          <State>
            {state === PomodoroTimerState.Pomodoro
              ? "🍅"
              : state === PomodoroTimerState.ShortBreak
                ? "🛌"
                : "🛌🛌"}
          </State>
          <TextButton onClick={() => pomodoroTimerStarter.start()}>
            Start
          </TextButton>
        </>
      ) : (
        <>
          <Time>
            <Minutes>{Math.floor(seconds / 60)}</Minutes>
            <Seconds>{seconds % 60}</Seconds>
          </Time>
          <TextButton onClick={() => pomodoroTimerStopper.stop()} secondary>
            Stop
          </TextButton>
        </>
      )}
    </Container>
  );
};
