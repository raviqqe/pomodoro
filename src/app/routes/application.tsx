import { styled } from "@linaria/react";
import { SignOut } from "../components/SignOut.js";
import { ViewGraphButton } from "../components/ViewGraphButton.js";
import { ViewTimerButton } from "../components/ViewTimerButton.js";
import { Outlet, useLocation } from "react-router";

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
`;

const Buttons = styled.div`
  position: fixed;
  top: 0.5em;
  right: 0.5em;

  > :not(:last-child) {
    margin-bottom: 0.5em;
  }
`;

export default (): JSX.Element => {
  const { pathname } = useLocation();

  return (
    <Container>
      <Outlet />
      <Buttons>
        <SignOut />
        {pathname === "/timer" ? <ViewGraphButton /> : <ViewTimerButton />}
      </Buttons>
    </Container>
  );
};
