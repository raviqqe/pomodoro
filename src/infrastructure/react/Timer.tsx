import styled from "styled-components";
import { PomodoroTimerState } from "../../application/pomodoro-timer-state";
import { TextButton } from "./TextButton";
import { white } from "./style/colors";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  > * {
    margin: 1rem;
  }
`;

const State = styled.div`
  font-size: 5em;
`;

const Time = styled.div`
  color: ${white};
  font-size: 10em;
`;

const Minutes = styled.span``;

const Seconds = styled.span`
  font-size: 0.6em;
  margin-left: 0.2ex;
`;

export interface IProps {
  seconds: number;
  startTimer: () => void;
  state: PomodoroTimerState;
  stopTimer: () => void;
  stopped: boolean;
}

export const Timer = ({
  seconds,
  startTimer,
  state,
  stopTimer,
  stopped,
}: IProps): JSX.Element => (
  <Container>
    {stopped ? (
      <>
        <State>
          {state === PomodoroTimerState.Pomodoro
            ? "🍅"
            : state === PomodoroTimerState.ShortBreak
            ? "🛌"
            : "🛌🛌"}
        </State>
        <TextButton onClick={startTimer}>Start</TextButton>
      </>
    ) : (
      <>
        <Time>
          <Minutes>{Math.floor(seconds / 60)}</Minutes>
          <Seconds>{seconds % 60}</Seconds>
        </Time>
        <TextButton onClick={stopTimer} secondary={true}>
          Stop
        </TextButton>
      </>
    )}
  </Container>
);
