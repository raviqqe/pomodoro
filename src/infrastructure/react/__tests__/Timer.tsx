import { create } from "react-test-renderer";
import { PomodoroTimerState } from "../../../application/pomodoro-timer-state";
import { Timer, IProps } from "../Timer";

const commonProps: Omit<IProps, "state" | "stopped"> = {
  seconds: 42,
  startTimer: async () => {},
  stopTimer: async () => {},
};

it("renders while running", () => {
  expect(
    create(
      <Timer
        {...commonProps}
        state={PomodoroTimerState.Pomodoro}
        stopped={false}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

it("renders with pomodoro state", () => {
  expect(
    create(
      <Timer
        {...commonProps}
        state={PomodoroTimerState.Pomodoro}
        stopped={true}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

it("renders with short break state", () => {
  expect(
    create(
      <Timer
        {...commonProps}
        state={PomodoroTimerState.ShortBreak}
        stopped={true}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

it("renders with long break state", () => {
  expect(
    create(
      <Timer
        {...commonProps}
        state={PomodoroTimerState.LongBreak}
        stopped={true}
      />
    ).toJSON()
  ).toMatchSnapshot();
});
