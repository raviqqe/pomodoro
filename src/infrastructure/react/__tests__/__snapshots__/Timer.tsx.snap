// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders while running 1`] = `
<div
  className="sc-dkPtRN kqnWZi"
>
  <div
    className="sc-eCImPb dWzspa"
  >
    <span
      className="sc-jRQBWg"
    >
      0
    </span>
    <span
      className="sc-gKclnd icXDZq"
    >
      42
    </span>
  </div>
  <button
    className="sc-bdvvtL sc-gsDKAQ gPIhRN djDaHe"
    onClick={[Function]}
  >
    Stop
  </button>
</div>
`;

exports[`renders with long break state 1`] = `
<div
  className="sc-dkPtRN kqnWZi"
>
  <div
    className="sc-hKwDye hGDmJA"
  >
    🛌🛌
  </div>
  <button
    className="sc-bdvvtL sc-gsDKAQ dsCIzG djDaHe"
    onClick={[Function]}
  >
    Start
  </button>
</div>
`;

exports[`renders with pomodoro state 1`] = `
<div
  className="sc-dkPtRN kqnWZi"
>
  <div
    className="sc-hKwDye hGDmJA"
  >
    🍅
  </div>
  <button
    className="sc-bdvvtL sc-gsDKAQ dsCIzG djDaHe"
    onClick={[Function]}
  >
    Start
  </button>
</div>
`;

exports[`renders with short break state 1`] = `
<div
  className="sc-dkPtRN kqnWZi"
>
  <div
    className="sc-hKwDye hGDmJA"
  >
    🛌
  </div>
  <button
    className="sc-bdvvtL sc-gsDKAQ dsCIzG djDaHe"
    onClick={[Function]}
  >
    Start
  </button>
</div>
`;
